### ISSUE TITLE
Repository Hooks do not trigger build

### ISSUE THREAD

**Issue by Shaoranlaos** [scm-manager/scm-jenkins-plugin#28]
### Bug Report

This happens in combination with the SCM-Manager Plugin in Jenkins where the build jobs are automatically created for each repository in a namespace folder.
This leads to a job path that is not handled by this plugin (/job/namespace/job/repo-name/job/branch name instead of /job/repo-name).

This seems to be only a problem for SVN repositories and not for git repos. Not sure why...
It is working if no repository specific config is done (so when the global config is used).

Workaround: setting the project name to namespace/job/repo-name/job/branch name (not tried yet if it is also working if I remove the branch name)

Also the CI-Status for SVN-Repos is not updated/received on the commit page but not sure if that is a problem on this side or in the SCM-Manager Plugin in Jenkins. It is working for GIT repos.

#### Describe how to reproduce the bug

Create a SVN-Repo in a namespace.
Configure the jenkins plugin directly on the repo with project = repo name.
Jenkins should automatically create the corresponding build job for the repository.
Make a commit in the repository.

--> Jenkins is not automatically starting a build job

If the repository specific config is removed the build is triggered.

#### Describe your SCM-Manager setup

SCM-Manager 2.20.0 with Jenkins plugin 2.2.1 behind a nginx reverse proxy
Jenkins 2.289.2 with the SCM-Manager Plugin 1.7.5 behind a nginx reverse proxy

#### Which operating system, browser and versions do you use?

Server with CentOS 7 (both SCM-Manager and Jenkins are running on the same server)

#### Log of the bug
The following Log comes for both Git and SVN repos if a repository specific configuration is added:
2021-07-26 14:26:38.507 [ScmEventBus-1-2] [          ] INFO  sonia.scm.trace.LoggingExporter - received failed Jenkins span, which took 1987ms:
 - url: https://######/ci/job/<job_name>/build?token=#####
 - method: POST
 - status: 404

**Comment by pfeuffer** [comment]
Hey, thanks for the report. We will take a look at it.

**Comment by pfeuffer** [comment]
Hi,

first, sorry for the long time that passed. Today I tried to reproduce your issue.

As far as I get it, the path pattern you mention (with the `/job/` elements in between) are not created by SCM-Manager, but are craeted by Jenkins, when you use automatic folders (you get the same layout when you create "Organization Folders" from GitHub).

The main difference between the global and the repository specific Jenkins settings in SCM-Manager is, that

- with a repository specific configuration a specific build is triggered. So here you would have to set the name using the `job` in between
![Screenshot-20220113172633-1032x298](https://user-images.githubusercontent.com/281097/149369244-e13d9f5f-004c-41e6-85a0-a6e5c78e8351.png)
- with the global config, SCM-Manager sends a general trigger naming the changed repository to Jenkins. Jenkins then checks, whether this repository is known and starts builds and pipeline updates.

So in short, you should use the global setting if you use folders. The repository specific setting should only be used for manually created build jobs with no further sub folders.

I hope I got your problem right. Otherwise feel free to tell me, what I got wrong. I will honestly try to respond more quickly next time.

Regards
René

**Comment by Shaoranlaos** [comment]
Hello,

thanks for coming back to the issue.

Ok then i will only us the global config. That works for me and the builds are correctly triggered.

I also mentioned a problem with the scm plugin in Jenkins (the build status is not correctly send to SCM-Manager from Jenkins for SVN-Repos), but i will file that on the jira issue page of the SCM-Manager plugin for Jira (because i think the problem is there and not on the SCM-Manager site because it works for Git-Repos).

BR

**Comment by pfeuffer** [comment]
Hi,

thanks for the feedback. The ci status looked ok for me, too (for both, the job from the namespace folder and the manual job). But of course we will take a look at your issue.

![Screenshot-20220114104838-1343x553](https://user-images.githubusercontent.com/281097/149495306-b83dcdd0-b14e-4737-a8d9-1ec619735c68.png)

René

**Comment by Shaoranlaos** [comment]
Link to Issue in Jenkins Jira: [JENKINS-67588](https://issues.jenkins.io/browse/JENKINS-67588)
